### PR TITLE
DMAEngine: Introduce Accelerate DMA.

### PR DESCRIPTION
**Pull request by FernandoS27**
This PR introduces DMA accelerations, this means doing a DMA Engine operation on the host GPU instead of the Guest GPU. Currently, the workflow is to Read memory and flush from host if needed, do a LLE DMA operation and then do Write that invalidates Host Caches. This is a very expensive operation on certain scenario.

This PR introduces DMA accelerations for Buffer Copies only. In the future more accelerations will be added for the BlockLinear-Pitch and back copies.

This PR fixes Shin Megami Tensei III Nocturne

### CODE REVIEW

**Comment by v1993** [comment]
This currently causes merge conflicts with Hades PR.

File: src/video_core/engines/maxwell_dma.cpp [hunk 1]
```diff
@@ -77,11 +80,29 @@ void MaxwellDMA::CopyPitchToPitch() {
...
+        // TODO: allow multisized components.
+        if (is_buffer_clear) {
+            ASSERT(regs.remap_const.component_size_minus_one == 3);
+            std::vector<u32> tmp_buffer(regs.line_length_in, regs.remap_consta_value);
+            memory_manager.WriteBlock(regs.offset_out, reinterpret_cast<u8*>(tmp_buffer.data()),
+                                      regs.line_length_in * sizeof(u32));
+            return;
+        }
+        UNIMPLEMENTED_IF(regs.launch_dma.remap_enable != 0);
+        if (!accelerate.BufferCopy(regs.offset_in, regs.offset_out, regs.line_length_in)) {
```
[the remark this answers is not in the document]
Comment (FernandoS27) [reply src/video_core/engines/maxwell_dma.cpp:96]: I don't want to couple more the rasterizer.

**Comment by v1993** [comment]
Probably untag this from EA until merge conflicts and build issues are fixed?

**Comment by gidoly** [comment]
fixes #5976, #4882

**Review by ameerj — APPROVED** [review]

**Pull request merged**